### PR TITLE
[node] Add headersDistinct and other 18.7.0 HTTP APIs

### PR DESCRIPTION
**Pull request by NotWoods**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://nodejs.org/api/http.html>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/node/http.d.ts [hunk 1]
```diff
@@ -479,11 +491,26 @@ declare module 'http' {
...
+         *
+         * If there were no previous value for the header, this is equivalent of calling `outgoingMessage.setHeader(name, value)`.
+         *
+         * Depending of the value of `options.uniqueHeaders` when the client request or the server were created,
+         * this will end up in the header being sent multiple times or a single time with values joined using `; `.
+         * @since v18.3.0, v16.17.0
+         * @param name Header name
+         * @param value Header value
+         */
+        appendHeader(name: string, value: number | string | ReadonlyArray<string>): this;
```
Comment (Semigradsky) [inline types/node/http.d.ts:513]: Isn't `value` a `string`? Why `number`?
https://nodejs.org/docs/latest-v18.x/api/http.html#outgoingmessageappendheadername-value

Comment (NotWoods) [reply types/node/http.d.ts:513]: I copied the definition from `setHeader`. Maybe both should be updated?

Comment (gabritto) [reply types/node/http.d.ts:513]: As pointed out by @Semigradsky, from the Node docs, `value` is `any` in `setHeader`, but should be `string | string[]` in `appendHeader`. Please update this accordingly.

**Review by gabritto — CHANGES_REQUESTED** [review]
This looks good except for the comment I left. Once that's fixed, I think this is good to merge.

**Review by gabritto — APPROVED** [review]

**Comment by NotWoods** [comment]
Ready to merge

**Pull request merged**